Source/JavaScriptCore/assembler/MacroAssemblerX86Common.h

```diff
@@ -905,7 +905,7 @@ class MacroAssemblerX86Common : public AbstractMacroAssembler<Assembler> {
         }
 
         // Add a negative using lea
-        m_assembler.leal_mr(WTF::negate(imm.m_value), src, dest);
+        m_assembler.leal_mr(-imm.m_value, src, dest);
```

Comment:
You need to handle `INT32_MIN` case. (In that case, you need to keep it as int32_min).

Reply:
Barring changing `WTF::negate` itself to make this work, let's just do safe two's complement here instead, something like `static_cast<int32_t>(~static_cast<uint32_t>(imm.m_value) + 1)`.